In this project i try to find those people who survied.<br>
By using csv file of titanic data.

In [1]:
#First import useful package 
# And cread dataframe to csv file
import pandas as pd
import numpy as np
# read_csv
url = r"C:\Users\ASHRAF\Downloads\titanic\train.csv"
df = pd.read_csv(url)
df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [4]:
# first copy this data frame:
df_train = df.copy()

<h1>1. first find missing values and get the idea about this dataframe.</h1>

In [6]:
print("the shape of this df:\n")
print(df_train.shape)
print("\n the info of data:\n")
print(df_train.info())
# find missing values:
print("\n is there any missing value availabe:\n",df_train.isnull().sum())

the shape of this df:

(891, 12)

 the info of data:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
None

 is there any missing value availabe:
 PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch       

<p>So i found 3 column which have lots of missing value.<br>
the <b>Cabin</b> column have majority number of missing values and i think it doesnot useful now.<br>
For Age and Embarked i will creat new column where the missing value fill with their median and mode.</p>

In [8]:
#For Age and Embarked column:
df_train['Age_Filled']=df_train['Age'].fillna(df_train['Age'].median())
df_train['Embarked_Filled']=df_train['Embarked'].fillna(df_train['Embarked'].mode())
df_train.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  Age_Filled Embarked_Filled  
0      0         A/5 21171   7.2500   NaN        S        22.0               S  
1      0          PC 17599  71.2833   C85        C        38.0               C  
2      0  STON/O2. 3101282   7.9250   NaN        S        26.0               S  
3      0            113803  53.1000  C123        S        35.0               S  
4      0            373450   8.0500   NaN        S        35.0               S  
5      0            330877   8.4583   NaN        Q        28.0               Q  
6      0             17463  51.8625   E46        S        54.0               S  
7      1            349909  21.0750   NaN        S         2.0               S  
8      2            347742  11.1333   NaN        S        27.0               S  
9      0            237736  30.0708   NaN        C        14.0               C

In [9]:
# drop the cabin column:
df_train.drop('Cabin',axis=1,inplace=True)
df_train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Embarked  Age_Filled Embarked_Filled  
0      0         A/5 21171   7.2500        S        22.0               S  
1      0          PC 17599  71.2833        C        38.0               C  
2      0  STON/O2. 3101282   7.9250        S        26.0               S  
3      0            113803  53.1000        S        35.0               S  
4      0            373450   8.0500        S        35.0               S

so as their gender(Sex) and Embarked string data.<br>
for ml it could be an obsticle so remove it by numerical data.

In [10]:
# convert male and female by 0 and 1:
df_train['Sex']=df_train['Sex'].map({'male':0,'female':1})
# convert C,Q,S by 0,1,2:
df_train['Embarked_Filled']=df_train['Embarked_Filled'].map({'C':0,'Q':1,'S':2})
print(df_train.head())

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    0  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    1  38.0      1      0   
2                             Heikkinen, Miss. Laina    1  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    1  35.0      1      0   
4                           Allen, Mr. William Henry    0  35.0      0      0   

             Ticket     Fare Embarked  Age_Filled  Embarked_Filled  
0         A/5 21171   7.2500        S        22.0              2.0  
1          PC 17599  71.2833        C        38.0              0.0  
2  STON/O2. 3101282   7.9250        S        26.0              2.0  
3            113803 

<h1>Explore Data(EDA)</h1>

In [11]:
# show the data of surviver by age:
# Note in survivour section 0=no and 1=yes.
print(df_train[df_train['Survived']==1]['Age'].describe())

count    290.000000
mean      28.343690
std       14.950952
min        0.420000
25%       19.000000
50%       28.000000
75%       36.000000
max       80.000000
Name: Age, dtype: float64


it shows that:<br>
1. 290 people survived.<br>
2. average age of people was 28.34<br>
3. spread of ages ≈ 15<br>
4. minimum age 0.42
5. 25% of age 19 .etc

In [12]:
# show which gender suvivied mostly.
print(df_train.groupby('Sex')['Survived'].mean())

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64


it show that female survived more than male.

In [13]:
# show the class of people survived:
print(df_train.groupby('Pclass')['Survived'].mean())

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


Note: pclass	:Ticket class	:1 = 1st, 2 = 2nd, 3 = 3rd